Source/JavaScriptCore/parser/Lexer.cpp

```diff
@@ -2519,8 +2519,18 @@ JSTokenType Lexer<T>::lexWithoutClearingLineTerminator(JSToken* tokenRecord, Opt
             shift();
             goto inSingleLineComment;
         }
-        // Otherwise, it could be a valid PrivateName.
-        if (isSingleCharacterIdentStart(next) || next == '\\') {
+
+        bool isValidPrivateName = false;
+        if (LIKELY(isLatin1(next)))
+            isValidPrivateName = isIdentStart(static_cast<LChar>(next)) || next == '\\';
+        else {
+            ASSERT(m_code + 1 < m_codeEnd);
+            char32_t codePoint;
+            U16_GET(m_code + 1, 0, 0, m_codeEnd - (m_code + 1), codePoint);
```

Comment:
Did you check that invalid and partial sequences cause `U16_GET` to give us a code point value that will return `false` for `isNonLatin1IdentStart`?

Reply:
My intention was to simply mimic what we do in order to get `CharacterIdentifierStart` for [the case  in which there is no preceding `#`](https://github.com/WebKit/WebKit/blob/main/Source/JavaScriptCore/parser/Lexer.cpp#L1926-L1938)—basically, if we _do_ find an unpaired surrogate, we expect that it therefore does not satisfy `ID_Start`.